### PR TITLE
Add first tests

### PR DESCRIPTION
**Pull request by rduckengineer**
- Setup a CLion environment (.idea/ folder)
- Added automatic savegame at the end of a timedemo
- Wrote unit tests for reading and writing 8 bit data in savegame file

### CODE REVIEW

File: src/doom/p_saveg.h [hunk 1]
```diff
@@ -62,5 +62,7 @@ void P_RestoreTargets (void);
 extern FILE *save_stream;
 extern boolean savegame_error;
 
+void  saveg_write8_in_stream  (FILE* stream, byte value , boolean* error);
+byte  saveg_read8_in_stream   (FILE* stream, boolean* error);
```
Comment (rduckengineer) [inline src/doom/p_saveg.h:66]: These are not meant to stay here, they might get removed / moved into a namespace later on

Comment (Cardellat) [reply src/doom/p_saveg.h:66]: Why not putting them in a namespace directly then?

File: src/doom/tests/savegame_test.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    FILE *test_file = fmemopen(buf.data(), 2*sizeof(byte), "a");
+    setbuf(test_file, NULL);
+
+    AND_GIVEN("A value to write")
+    {
+      byte value = 0x43;
+
+      AND_GIVEN("No prior error")
+      {
+        boolean err = false;
```
Comment (rduckengineer) [inline src/doom/tests/savegame_test.cpp]: I think those two could be collapsed together
```suggestion
    AND_GIVEN("A value to write and no prior error")
    {
      byte value = 0x43;
      boolean err = false;
```

File: src/doom/p_saveg.h (hunk 1 above)
Comment (rduckengineer) [reply src/doom/p_saveg.h:66]: Because the engine files are still in C, and there's no namespace in C

**Pull request merged**